### PR TITLE
LPS-198950 Modal to update existing object definition is being shown for invalid file

### PR DESCRIPTION
**Pull request by Artartur**
Related Issue: [LPS-198950](https://liferay.atlassian.net/browse/LPS-198950)

### CODE REVIEW

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalImportObjectDefinition.tsx [hunk 1]
```diff
@@ -67,6 +67,15 @@ export default function ModalImportObjectDefinition({
 		},
 	});
 
+	const cleanData = () => {
+		setError(Liferay.Language.get('the-structure-failed-to-import'));
+		setExternalReferenceCode('');
+		setFile({
+			fileName: '',
+			inputFile: null,
+		});
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalImportObjectDefinition.tsx:76]: Hey @Artartur, very good solution! As this function will only be used in catch, you can put all of this inside catch, making the code easier to read:

```tsx
	catch (error) {
		setError(Liferay.Language.get('the-structure-failed-to-import'));
		setExternalReferenceCode('');
		setFile({
			fileName: '',
			inputFile: null,
		});
	}
```

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalImportObjectDefinition.tsx [hunk 2]
```diff
@@ -261,7 +270,7 @@ export default function ModalImportObjectDefinition({
 										);
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalImportObjectDefinition.tsx:270]: @Artartur one more thing, here we have to clear the error here, for example, if the user imports for the first time a structure that fails in the import process, we will show the error, but if after that he imports a valid structure the error will remain on the screen causing a wrong feeling then we have to clear this error after successful attempt:

```tsx
setError('');
```

**Review by danvb112 — CHANGES_REQUESTED** [review]
Hey @Artartur, good work on this bug! I have some suggestions :smile:

**Comment by Artartur** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalImportObjectDefinition.tsx (hunk 1 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalImportObjectDefinition.tsx]: Done!

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalImportObjectDefinition.tsx (hunk 2 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalImportObjectDefinition.tsx:262]: I'm not sure if I've added it in the right place, can you confirm?

Comment (danvb112) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalImportObjectDefinition.tsx:262]: @Artartur actually you have to put this `setError('');` here, inside this `try` statement after `setExternalReferenceCode`

**Review by danvb112 — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalImportObjectDefinition.tsx (hunk 2 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalImportObjectDefinition.tsx:261]: Done!

**Comment by Artartur** [comment]
ci:test:sf

**Review by danvb112 — CHANGES_REQUESTED** [review]
@Artartur Good work! One more thing, if we close the modal with the error message and open it again, the message still remains in the modal, so we also have to clear the message (`setError('');`) inside the `onClose` function on line `58`. Sorry I didn't see this in my last review.

**Comment by Artartur** [comment]
ci:test:sf

**Review by alinedoleron — APPROVED** [review]
LGTM!

**Comment by alinedoleron** [comment]
ci:forward

**Pull request closed without merging**